[v1.5rc][W13-B4] PrioriTask

dominickenn opened the pull request.

dominickenn committed:
DeveloperGuide: Rework UML diagrams

aguss787 committed:
Refactor: PersonCard -> TaskCard

aguss787 committed:
Refactor: PersonListPanel -> TaskListPanel

aguss787 committed:
Refactor: PersonListCard -> TaskListCard

aguss787 committed:
Refactor: personListView -> taskListView

aguss787 committed:
Refactor: PersonListViewCell -> TaskListViewCell

aguss787 committed:
Refactor: PersonPanelSelectionChangedEvent -> TaskPanelSelectionChangedEvent

aguss787 committed:
Reformat

dominickenn committed:
Merge pull request #9 from aguss787/aguss787-refactor-ui

Refactor UI

guekling committed:
Merge pull request #15 from dominickenn/DeveloperGuide

DeveloperGuide: Rework UML diagrams

dominickenn committed:
[#1]Refactor all class names to support organizer (#17)

* Refactor : Storage
AddressBook -> Organizer
Person -> Task

* Refactor: Model

* Refactor: Commons

* Refactor: testutil

* Refactor: AddressBookParser -> OrganizerParser

* Style fix

guekling committed:
DeveloperGuide v1.0: Edit Target User Profile, Value Proposition & NFR (#18)

* Edit target user profile, value proposition & NFR

* Edit value proposition for grammatical errors

natania-d committed:
[#6] DeveloperGuide: Edit Use Case (#20)

* Developer Guide: edit use case

* Developer Guide: change TaskBook to Organizer

guekling committed:
[#3, #8] UserGuide v1.0: Rework Introduction, Quick Start & Command Summary sections (#16)

* Rework commands in Features & Command Summary sections

* Rework Introduction, Quick Start & Command Format

* Change TaskBook to Organizer

aguss787 committed:
Refactor: address -> organizer (#19)

* Refactor address->organizer

* Reformat

dominickenn committed:
Logic minor enhancement (#22)

* Add command_alias for clear

* Allow OrganizerParser to parse ClearCommand#COMMAND_ALIAS

* Add test case for ClearCommand#COMMAND_ALIAS

* Add ClearCommand#COMMAND_ALIAS to UserGuide

* Add COMMAND_ALIAS for AddCommand

* Allow OrganizerParser to parse AddCommand#COMMAND_ALIAS

* Add testcase for AddCommand#COMMAND_ALIAS

* Add AddCommand#COMMAND_ALIAS to UserGuide

* Add COMMAND_ALIAS for DeleteCommand

* Allow OrganizerParser to parse DeleteCommand#COMMAND_ALIAS

* Add testcase for DeleteCommand#COMMAND_ALIAS

* Add DeleteCommand#COMMAND_ALIAS to UserGuide

* Add COMMAND_ALIAS for EditCommand

* Allow OrganizerParser to parse EditCommand#COMMAND_ALIAS

* Add testcase for EditCommand#COMMAND_ALIAS

* Add EditCommand#COMMAND_ALIAS to UserGuide

* Add COMMAND_ALIAS for FindCommand

* Allow OrganizerParser to parse FindCommand#COMMAND_ALIAS

* Add testcase for FindCommand#COMMAND_ALIAS

* Add FindCommand#COMMAND_ALIAS to USERGUIDE

* Add COMMAND_ALIAS for ListCommand

* Allow OrganizerParser to parse ListCommand#COMMAND_ALIAS

* Add testcase for ListCommand#COMMAND_ALIAS

* Add ListCommand#COMMAND_ALIAS to UserGuide

* Add COMMAND_ALIAS for SelectCommand

* Allow OrganizerParser to parse SelectCommand#COMMAND_ALIAS

* Add testcase for SelectCommand#COMMAND_ALIAS

* Add SelectCommand#COMMAND_ALIAS to UserGuide

dominickenn committed:
[#6][#13] Update documentation (#23)

* Update README
Travis Link update
Addressbook -> PrioriTask

* Add UserStories to DeveloperGuide

* Update UML diagrams and corresponding images

* Rename image files and update DeveloperGuide

guekling committed:
[#25] Change UI Component: Use different colours for different tags inside task cards (#24)

* Add 10 new colours for labels

* Assign different colours to different tags

* Fix checkstyle errors

* Fix checkstyle errors

* Add tests

* Add credit for reused work

guekling committed:
[#26] Change UI Component: Modify the StatusBarFooter to show the total number of tasks in the organizer (#27)

* Update StatusBarFooter to show total number of tasks in Organizer

* Add tests

* Fix mistake in ClearCommandSystemTest.java

* Fix Travis failures

guekling commented:
@aguss787 @natania-d 

Changes Made: 
- Developer Guide: Target User Profile, Value Proposition, NFR, User Stories & Use Cases
- User Guide: Features & Command Summary (AddressBook -> Organizer)
- UML Diagrams (AddressBook -> Organizer)
- Refactoring from AddressBook -> Organizer

Enhancements done for `Logic`, `Model` and `UI` components.

natania-d committed:
Model enhancement (#68)

* Implement a removeTag(Tag) method and refactor 'person' to 'task'

* Implement removeUnusedTags() method.

* Add a new API method deleteTag(Tag) in ModelManager and refactor methods with 'person' to 'task'

* Add new tests for removeTag(Tag) method and deleteTag(Tag) method.

* Debug in AddCommandTest.java

brandonyeoxg commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,51 +782,118 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 *Target user profile*:
 
-* has a need to manage a significant number of contacts
+* university students
```
Recall from tutorial that this target user is rather vague and generic, perhaps narrowing of the scope to a more focused target user would help maintain the workload on the team to produce a more feature complete product at the end of v1.5rc =). 

It seems that B3 has the same target audience as this... perhaps the team can look into differentiating from B3? =P

brandonyeoxg commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,51 +782,118 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 [appendix]
 == User Stories
 
+The following is a list of user stories for PrioriTask.
```
Is it `Organizer` or `PrioriTask`? =P

guekling replied:
We use `Organizer` as our storage, but `PrioriTask` as our product name. (:

brandonyeoxg commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,51 +782,118 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+
+|`* *` |user |edit a task's description|change a task's description
+
+|`* * *` |user |mark a task as complete |indicate to the application to push a task to the completed list
```
Perhaps `indicate to the application to push a task to the completed list` is not appropriate for the `so that I can` portion as it indicates what the application is supposed to do and not how this feature solves for the user.

Would `complete my task` fit better?

brandonyeoxg commented on docs/DeveloperGuide.adoc:
```diff
@@ -782,51 +782,118 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+
+|`* *` |user with many tasks in the to-do list |find a task by description|locate a task when I have forgotten it's name
+
+|`* *` |user |have all tasks' priority levels be automatically updated as the deadline approaches |transfer the responsibility of updating priorities to the application
```
Same as the comment provided above, it seems that the `So that I can` portion does not flesh out the solution to the user's problem.
Instead the user story at line 863 seems to also solve the user's problem for line 861, perhaps both user stories can be collapsed into 1 user story?

brandonyeoxg commented on docs/DeveloperGuide.adoc:
```diff
@@ -850,8 +917,11 @@ _{More to be added}_
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 tasks without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should come with a help feature and various helper messages so that it is usable by a novice who had only have minimal CLI experience.
```
Good NFR!

brandonyeoxg commented on src/main/java/seedu/organizer/model/Organizer.java:
```diff
@@ -104,15 +104,27 @@ public void addPerson(Task p) throws DuplicateTaskException {
+    }
+
+    /**
+     * Removes all {@code Tag}s that are not used by any {@code Task} in this {@code Organizer}.
```
Concise header comments!

brandonyeoxg commented on src/main/java/seedu/organizer/model/Organizer.java:
```diff
@@ -156,17 +168,53 @@ public void addTag(Tag t) throws UniqueTagList.DuplicateTagException {
+    /**
+     * Removes {@code tag} from all tasks in this {@code Organizer}.
+     */
+    public void removeTag(Tag tag) {
```
Can we improve this code piece by adding assert(s) that we discussed in the tutorial?

brandonyeoxg reviewed (commented):
@dominickenn @guekling @aguss787 @natania-d Some comments added. 
**DO NOT CLOSE THIS PR** =), we will iterate on and update Organizer through this PR!

guekling committed:
[#13] Update README (#69)

* Add AppVeyor & Coveralls badges, Update Ui.png

* Update README

* Fix styling issues

* Attempt to add blank line

* Revert previous commit

* Update AppVeyor badge

guekling committed:
[#11] Update About Us (#70)

* Update Project Team, Add pictures

* Add Project Portfolio doc for all members

* Add newline at EOF

* Rectify Dominic responsibility

dominickenn committed:
[#71][#73][#74][#76][#28][#41] Phone to Priority refactor (#77)

* Refactor Task#Phone to Task#Priority

* Fix TypicalTasks

* Fix validOrganizer.xml

* Fix testcase

* Add lowest priority level

* Refactor Task#Phone to Task#Priority

* Fix TypicalTasks

* Fix validOrganizer.xml

* Fix testcase

* Add lowest priority level

* Stylecheck fix

guekling committed:
[#72] Refactor Email -> Deadline (#75)

* Add new Deadline class

* Refactor Task#Email -> Task#Deadline in Task.java

* Refactor Task#Email -> Task#Deadline in ParserUtil.java

* Refactor Task#Email -> Task#Deadline in XmlAdaptedTask.java

* Refactor Task#Email -> Task#Deadline in EditCommand.java & CliSyntax.java

* Fix typo

* Refactor Task#Email -> Task#Deadline in AddCommandParser.java

* Refactor Task#Email -> Task#Deadline in SampleDataUtil.java

* Refactor Task#Email -> Task#Deadline in AddCommandParserTest.java and related classes

* Refactor Task#Email -> Task#Deadline in EditCommandParserTest.java

* Renamed EmailTest.java -> DeadlineTest.java and refactor Task#Email -> Task#Deadline

* Refactor Task#Email -> Task#Deadline in DeadlineTest.java

* Refactor Task#Email -> Task#Deadline in XmlAdaptedTaskTest.java

* Refactor remaining Task#Email -> Task#Deadline in XmlAdaptedTaskTest.java and related classes

* Refactor remaining Task#Email -> Task#Deadline in AddCommandSystemTest & EditCommandSystemTest

* Refactor missed Task#Email -> Task#Deadline in AddCommandSystemTest & EditCommandSystemTest

* Refactor remaining Task#Email -> Task#Deadline in ParserUtilTest.java

* Delete Email class

* Fix typo

* Refactor Email -> Deadline in related files

* Fix checkstyle errors

* Refactor missed Email -> Deadline

* Refactor UI tests from Email -> Deadline

* Refactor missed Email -> Deadline in FindCommandSystemTest

* Fix checkstyle errors

* Fix checkstyle error

* Fix test issues

* Update comment in Deadline class

* Change label from Email -> Deadline for TaskCard.fxml

* Fix test issues

* Add date to example in AddCommand.java

guekling committed:
[#78] Refactor Address -> Description (#79)

* Add new Description class

* Fix typo error in Deadline.java

* Change Address -> Description in Task.java

* Change Address -> Description in ParserUtil.java

* Change Address -> Description in XmlAdaptedTask.java

* Change Address -> Description in EditCommand.java & CliSyntax.java

* Change Address -> Description in AddCommandParser.java

* Change Address -> Description in SampleDataUtil.java

* Change getSampleAddressBook() -> getSampleOrganizer()

* Change addressBookOptional -> organizerOptional

* Change Address -> Description in CommandTestUtil.java & EditTaskDescriptorBuilder.java

* Change Address -> Description in AddCommandParserTest.java & TaskBuilder.java

* Change Address -> Description in EditCommandParserTest.java

* Change Address -> Description in XmlAdaptedTaskTest.java

* Change Address -> Description in AddCommandSystemTest.java

* Change Address -> Description in EditCommandSystemTest.java

* Change Address -> Description in ParserUtilTest.java

* Rename AddressTest.java -> DescriptionTest.java, Update tests accordingly

* Change Address -> Description in TaskListCard.fxml

* Change Address -> Description in TaskCard.java

* Delete Address class

* Change Address -> Description in AddCommand.java & EditCommandParser.java

* Update example for description

* Change Address -> Description in Organizer.java

* Fix checkstyle errors

* Change Address -> Description in TaskUtil.java

* Change Address -> Description in EditTaskDescriptorTest.java

* Change Address -> Description in TypicalTasks.java

* Change Address -> Description in NameContainsKeywordsPredicateTest.java

* Change Address -> Description in TaskCardHandle.java & GuiTestAssert.java

* Change Address -> Description in FindCommandSystemTest.java

* Update validOrganizer.xml data

* Fix checkstyle errors for tests

* Update XML files

* Change DESC_AMY & DESC_BOB to DESC_EXAM & DESC_STUDY

guekling committed:
Rename images to follow naming conventions (#80)

guekling committed:
[#81] DeveloperGuide: Update Target User Profile & User Stories (#82)

* Update Target User Profile

* Fix English errors

* Fix User Stories according to PR comments

dominickenn committed:
[#83][#84][#85][#60]  Add automatic priority level setting to addCommand (#86)

* Allow task to be added without priority
Priority level is set to 0 if no p/ prefix is found

* Add test for new addCommand functionality

* Add documenation in UserGuide

* Fix merge conflicts

dominickenn committed:
[#91][#92][#90] Add automatic task sorting according to priority level (#93)

* Add autosorting of tasks function to Organizer

* Edit tests to support autosort

* Add and fix tests for new autosort function

* Add UserStory for new autosort function

* DeveloperGuide typo fix

guekling committed:
[#87] Make Description a non-compulsory field (#94)

* Edit comments

* Make Description a non-compulsory param

* Update comment for Description param

* Modify DescriptionTest

* Edit & Add tests accordingly

* Remove unnecessary lines

* Update comment according to deleted lines

* Add test for Description class

* Fix checkstyle errors

* Add credit

guekling committed:
Update UserGuide Features, Fix some naming & English issues (#96)

guekling committed:
[WIP] [#97] Create new FindDescription command (#98)

* Create new DescriptionContainsKeywordsPredicate class

* Update MessageUsage for FindCommand

* Add new FindDescriptionCommand class

* Add new FindDescriptionCommandParser.java

* Modify FindCommand & OrganizerParser to reflect changes accordingly

* Refactor FindCommand -> FindNameCommand

* Fix checkstyle errors

guekling committed:
Revert "[WIP] [#97] Create new FindDescription command (#98)" (#99)

This reverts commit 983010ab6afa3eaa543dd77b5a0de3e506734f29.

aguss787 committed:
[#45] Add status and toggle status command (#101)

* Add status

* Update testcase

* Edit ui

* Fix styling

* Add testcase for Status

* Add toggle parser rest

* ADd test for ToggleCommand

* Update OrganizerParserTest

* Update user guide

guekling committed:
[#97] Create new FindDescription command (#100)

* Create new DescriptionContainsKeywordsPredicate class

* Update MessageUsage for FindCommand

* Add new FindDescriptionCommand class

* Add new FindDescriptionCommandParser.java

* Modify FindCommand & OrganizerParser to reflect changes accordingly

* Refactor FindCommand -> FindNameCommand

* Fix checkstyle errors

* Add test for FindDescriptionCommand in OrganizerParser

* Add credit

* Add new FindDescriptionCommandTest file

* Create new abstract class FindCommand to prevent duplicate codes

* Fix checkstyle and test issues

* Add tests for FindDescriptionCommandParser

* Add tests for DescriptionContainsKeywordsPredicate

* Fix checkstyle issues

* Rename FindCommand to FindCommandTest

* Add credit

* Update UserGuide for FindName & FindDescription command

guekling committed:
[#89] Make Deadline a non-compulsory field (#102)

* Modify Deadline & AddCommandParser to make deadline non-compulsory field

* Add credit

* Add tests

* Modify tests in AddCommandSystemTest

* Add test to DeadlineTest

dominickenn committed:
[#104] Task priority level comparison fix (#105)

* Task comparator fix

* Tests fix

guekling commented:
**Week 8**

- `Priority List` field
  Sort tasks by `Priority List`
- `Description` field
  `FindDescription` command
- `Deadline` field
- `ToggleCommand`

brandonyeoxg commented:
@dominickenn @guekling @aguss787 @natania-d Hey guys do attend the CS2103t lecture at 4pm today to get a better understanding on what constitutes to a major/minor feature =).

guekling committed:
[#107] Save Deadline into a LocalDate variable (#114)

* Save deadline's date to LocalDate variable

* Make Deadline a compulsory field

* Fix tests errors

* Add more tests

* Fix checkstyle errors

* Edit tests

* Fix checkstyle errors

* Attempt at increasing Coveralls %

* Another attempt at increasing Coveralls %

guekling committed:
[#112] Create new FindMultipleFieldsCommand command (#113)

* Create new FindCommand command

* Edit comments for FindDescriptionCommand & FindNameCommand

* Modify FindCommand by using FieldContainsKeywordsPredicate class

* Add tests to OrganizerParserTest

* Fix typo error in FindNameCommandTest

* Rename FindCommand -> FindMultipleFieldsCommand

* Add tests to FindMultipleFieldsCommandTest

* Add credits

* Add tests for FindMultipleFieldsCommandParserTest

* Add tests in MultipleFieldsContainsKeywordsPredicateTest

* Modify test in NameContainsKeywordsPredicateTest

* Fix typo errors

* Fix checkstyle errors

* Update UserGuide with new command

* Rename all find commands command_word

* Add credits

brandonyeoxg commented on docs/UserGuide.adoc:
```diff
@@ -63,33 +82,34 @@ Format: `help`
-
-[TIP]
-A task can have any number of tags (including 0)
+Adds a person to the PrioriTask +
```
Isn't the add command supposed to add a task? Why is a person added instead =p.

brandonyeoxg commented on docs/UserGuide.adoc:
```diff
@@ -146,20 +206,40 @@ Selects the task identified by the index number used in the last task listing. +
 `select 1` +
 Selects the 1st task in the results of the `find` command.
 
+=== Mark task as complete: `toggle` or `t`
```
Would having a command `complete` or `done` be more intuitive to users than `toggle`?

brandonyeoxg commented on docs/UserGuide.adoc:
```diff
@@ -232,14 +312,60 @@ Format: `exit`
+
+_{ coming in v2.0 }_
+
+=== Locating tasks by description `[coming in v2.0]`
```
Isn't locating task by description done by the `finddes` command stated above? =p

brandonyeoxg commented on docs/UserGuide.adoc:
```diff
@@ -248,19 +374,23 @@ _{explain how the user can enable/disable data encryption}_
 e.g. `edit 1 p/9 d/2018-12-30`
-* *Find* : `find KEYWORD [MORE_KEYWORDS]` +
-e.g. `find Developer User`
+* *Find Name* : `findn KEYWORD [MORE_KEYWORDS]` or `fn KEYWORD [MORE_KEYWORDS]` +
```
Where is the `Find` command summary to find both des and name? =0

brandonyeoxg reviewed (commented):
@dominickenn @guekling @aguss787 @natania-d 

Some admin matters.
I would want you to mention the following in your **Developer guide**
1. For each team member what are the features (major and minor) you are proposing?
2. Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

dominickenn committed:
Documentation update (#121)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

guekling committed:
[#103] Add new FindDeadline command (#118)

* Add new FindDeadline command

* Add tests

* Fix checkstyle errors

* Update UserGuide

* Edit MultipleFieldsComand to be able to search for deadline

* Modify FindMultipleFieldsCommand tests

* Update UserGuide

dominickenn committed:
Refactor terms in DeveloperGuide and fix typo in UserGuide [#109][#110] (#122)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Fix title

* Fix image

* Refactor Design and Implementation

dominickenn committed:
[#106][#115][#116][#117][#55][#56] Add DateAdded to Task. Add priority autoupdating feature. (#123)

* Add Priority#HIGHEST_SETTABLE_PRIORITY_LEVEL

* Fix collating code

* Add updatePriority function

* Add autoupdate of priority function during addition or setting of tasks

* Fix automatic update of priority level

* UGDG update of new function

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Add Priority#HIGHEST_SETTABLE_PRIORITY_LEVEL

* Fix collating code

* Add updatePriority function

* Add autoupdate of priority function during addition or setting of tasks

* Fix automatic update of priority level

* UGDG update of new function

* Fix editTask

* Add Task#DateAdded

* Test fix

* Stylecheck fix

* Merge conflict fix

* Add comments and update if statement

* Add collation code

* Bug fix for priorityupdate condition

* Add tests for DateAdded and Priority AutoUpdate

dominickenn committed:
[#110] Add priority implementation documentation (#124)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Edit docmentation of Natania major/minor feature

* Fix undo/redo implementation documentation

* Add figure reference

* Fix static method

* Add diagrams and images

* Add implementation of priority documentation

* Test fix

dominickenn committed:
[#125] Test bug fix (#127)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Temporary bug fix and test fix

guekling committed:
Update DeveloperGuide (#133)

Add calendar implementation in dev guide

aguss787 committed:
Add Subtask (#131)

* Add subtask

* Add addSubtaskCommand

* Update used guide

brandonyeoxg commented:
@dominickenn @guekling @aguss787 @natania-d regarding the `Implementation` section in the dev guide:
The `Task priority auto updating and sorting` feature is informative with a sequence diagram and several other diagrams provided =).
Perhaps we can make it more readable by improving on the formatting of the content, maybe a sub heading for auto updating of priority and a sub heading for the sorting of the priority. 
This will help the reader quickly identify which portion of the feature he is looking at.

The `Calendar` feature is likewise informative with class diagram provided and a general fxml representation. Perhaps for the class diagram, we can omit the member variables and methods that are not used in the discussion of the implementation, since methods like `setFiveWeeksMonthCalendar` is not included in the discussion.

Overall good effort for the DG and remember to update the developer guide and user guide as you go along, so keep it up. Remember to update your PR title.

One thing I forgot to share in the tutorial is that each of you should maintain a Project Portfolio document individually which may be used for grading at the end of the course. Thus you can start extracting your individual portion of documentation in both UG and DG. **Note: Project Portfolio is not a requirement for v1.3.**

guekling committed:
[#109] Update UserGuide (#146)

* Update UserGuide

* Attempt to edit CSS

* Another attempt at using CSS

* Another attempt at figuring out CSS classes

* Still attempting the same thing

* Still attempting to do it...

* I think it's finally working

* Making NoteBlocks

* Trying to figure out how to override CSS

* Fix typo error

* Adding more levels

* Attempt to add icons

* Another attempt at adding icons

* Giving up on borders

* Still fixing the Note

* Still fixing the Note

* Finish up Note, Add Format block

* Edit Format box

* Add icon for Format

* Fix typo

* Trying Format block on another

* Add Format box to all commands

* Fix typo

* Adding blank lines

* Edit CSS

* Format more Format boxes

* Padding right

* Previous commit didn't work - change CSS

* Examples

* Change header colors

* Add some Example

* Add all Example

* Edit header colors

* Edit header styles

* Example block

* Example Block

* Example Icon

* Change Example Icon

* Add all Example Box

* Add border to example code

* Format code for Example and Format

* Wrong color

* Add more classes

* Remove some exampleblocks

* Remove exampleblocks for all

* Change CSS file name from user-guide.css -> guides-style.css

* Add Important blocks

* Change header colors

* Update Subtask

* Empty commit to test Travis

* Fix checkstyle errors

guekling committed:
[#108] Add Month Calendar View (#145)

* Calendar layout in BrowserPanel.fxml

* Comment out unused code with Google search engine

* Show current month & year as title

* Calendar view for a single month

* Abstract code in BrowserPanel.java & write comments for methods

* Browser Panel & MonthView working together successfully

* Browser Panel, Calendar & MonthView working successfully

* BrowserPanel -> CalendarPanel

* Fix checkstyle issues and comment out tests

* Rename BroserPanelHandle -> CalendarPanelHandle, Comment out more UI tests

* Comment out more tests in HelpCommandSystemTest

* Add comments

* Delete DummySearchPage

* Change Calendar folder name

* Empty commit to test Travis

* Attempt to solve Travis issues

* Edit current methods and add new methods

* Fix checkstyle issues

aguss787 committed:
Add subtask implementation (#148)

aguss787 committed:
Fix fullstop inconsistency (#149)

guekling committed:
[#110] Update DeveloperGuide (#150)

* Fix issues that was brought up in class + a few other minor ones

* Fix TA issues

* Fix Java code blocks

* Remove Suggested Programming Tasks to Get Started

* Fix minor issues

* Trying to fix Java code blocks

* Attempt at fixing Java code blocks

guekling committed:
[#153] View Tasks on Calendar (#154)

* Create ListView in MonthView

* Create EntryCard, works with MonthView

* Sort by deadline working

* FilteredList & SortList working in MonthView

* Entries are being populated onto the calendar properly

* Entries are being updated

* Add comments

* Fix typo error.

aguss787 committed:
Add toggle subtask (#155)

* Add toogle subtask

* update userguide

* fix user guide for toggle subtask

aguss787 committed:
Add delete subtask command (#157)

* Add delete subtask command

* Add userguide for delete subtask

* Add endline

natania-d committed:
Added DateCompleted parameter (#159)

* build gradle

* Create new DateCompleted class

* Revert "build gradle"

This reverts commit c6a9a19619f026d00446bc23869d5b51dca2b56b.

* Create new DateCompleted class

* Edit DateCompleted

* Add DateCompleted parameter logic to ToggleCommand. Temporary fix for UI.

* Fixed UI to show DateCompleted

* Fixed bug for DateCompleted

* Fixed checkstyle errors

* Fixed checkstye error

* Fixed compileTestJava errors

* Fix checkstyleTest Error

* Add DateCompletedTest

* Fixed test errors

* Fix test errors

* Update version number in MainApp.java

* Revert "Update version number in MainApp.java"

This reverts commit 245cc572a9fd914acf233b1318610e956816079e.

guekling committed:
[#158] Add PreviousMonthCommand (#165)

* New PreviousMonthCommand.java

* Update OrganizerParser

* Modifying CalendarPanel & Calendar accordingly - failed

* Remove unused import statement

* Calendar.java is useless, PrevMonthButton works in MonthView.java

* Command is working

* Remove Calendar.java class

* PreviousMonthCommand works successfully; Minor bugs

* Add comments

* Fix typo error

* Fix bugs in MonthView, Add comments

* Remove Logger from CalendarPanel

* Add comments in Logic

* Fix checkstyle issues

* Add test to OrganizerParserTest

* Add credit

* Remove unnecessary line in LogicManager

* Refactor add method to multiple smaller methods in AddCommandSystemTest

* Fix checkstyle issues

* Refactor add method to multiple smaller methods in EditCommandSystemTest

* Fix checkstyle issues

* Refactor add method to multiple smaller methods in FindNameCommandSystemTest

* Update UserGuide

* Add comment for PreviousMonthCommand

* Empty commit

* Fix checkstyle issue

guekling committed:
[#147] Calendar Tests (#167)

* Add EntryCardHandle & EntryCardTest

* Modify CalendarPanelHandle & CalendarPanelTest, Add new getter method in CalendarPanel

* Add new equals() method in EntryCard, Updated test

* setId to ListView & Text, add Platform.runLater

* Add new MonthViewHandle & MonthViewTest

* Fix checkstyle issues

* Add pause() in GuiRobot

* Update test in MonthViewTest

* Add more tests in MonthView

* Add PreviousMonthCommandTesT

* Add test for equals()

* Add more tests for equals()

* Add guiRobot.pause()

* Clean up

* Empty commit

dominickenn committed:
[#129][#130][#136][#137][#139][#140] Add signUpUser and loginUserCommand (#168)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Add user stories

* Add user class

* Add UniqueUserList

* Update UniqueUserList

* Add user to task

* Add XmlAdaptedUser

* Add UniqueUserList to Organizer

* Fix XmlAdaptedTask

* Fix tests

* Add SignUpCommand

* Fix and create tests

* Merge and test fix

* Add feature to only allow command when user is logged in

* Add TaskByUserPredicate

* Rework clear command

* Fix tests

* Add tests

* Add tests

* Add tests

brandonyeoxg commented on src/main/java/seedu/organizer/logic/commands/ToggleCommand.java:
```diff
@@ -19,7 +23,7 @@
 
 /**
- * Inverse the value of task status
+ * Inverse the value of task status (Done or Not done)
```
Perhaps this comment should also clarify if subtasks are also considered done or not based on the main task's completion? =P

brandonyeoxg commented on src/main/java/seedu/organizer/logic/commands/ToggleSubtaskCommand.java:
```diff
@@ -0,0 +1,122 @@
+import seedu.organizer.model.user.User;
+
+/**
+ * Inverse the value of task status
```
Perhaps you meant subtask?

brandonyeoxg commented on src/main/java/seedu/organizer/logic/commands/ToggleSubtaskCommand.java:
```diff
@@ -0,0 +1,122 @@
+    private Subtask editedSubtask;
+
+    /**
+     * @param taskIndex index of the task in the filtered task list to edit
```
The start of the header comment does not follow the convention (i.e. first sentence should be something like `Selects next line …` instead of `Select next line` or `This method is for selecting …`)

brandonyeoxg commented on src/main/java/seedu/organizer/model/task/DateCompleted.java:
```diff
@@ -0,0 +1,108 @@
+    /**
+     * Constructs a DateCompleted that is empty
+     */
+    public DateCompleted(boolean completed) {
```
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. name for boolean doesn't follow naming convention

brandonyeoxg reviewed (commented):
@dominickenn @guekling @aguss787 @natania-d just some quick comments on the UG and DG.

UG:
So far the UG is fairly comprehensive at this milestone, just some suggestions to refine it further:
1) Perhaps in Section `3` we can go through the parameters of the subtask to make it more consistent?
2) The order of `Subtask Management Features` in section `4` could perhaps be rearranged? I almost overlooked it as it came after features that are `[coming in v2.0]` and maybe the other users would overlook too during the mock test? Similarly for the `Calendar Features`.

DG:
1) Do fill up the placeholders like `{ Introduction to be added. }`
2) Do edit portions like `2.4.2` to be inline with PrioriTask  =).

dominickenn committed:
[#138][#141] Add currently logged in user filtering (#170)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Stylecheck fix

* Fix typo

* Add current user task filtering

* Fix tests #1

* Fix tests #2

* Fix tests #3

* Fix tests #4

* Fix tests #5

* Revert "Fix tests #5"

This reverts commit 4470790a5504b067eae8ace25b3f0d18637fc182.

* Revert "Fix tests #4"

This reverts commit b066a9a88697b50d8d9ae4a61ae8e76075c49e56.

* Fix tests #6

* Fix tets #7

* Revert "Fix tets #7"

This reverts commit 448c78d4ca92d89f6b3ad47fba7d25aab1e2ce38.

* Comment out some tests due to travis speed.
Commented out :
CalendarPanelTest : line 71, problem with getExpectedRowColumn() method
AddCommandSystemTest : local tests pass, but travis fails as its too slow
ClearCommandSystemTest : local tests pass, but travis fails as its too slow
DeleteCommandSystemTest : local tests pass, but travis fails as its too slow
EditCommandSystemTest : local tests pass, but travis fails as its too slow
FindNameCommandSystemTest : local tests pass, but travis fails as its too slow

dominickenn committed:
[#152][#177] Add logoutCommand (#178)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Remove getSampleData feature

* Add LogoutCommand

* Uncomment tests

* Add tests

* Add documentation of logout

* Fix checkstyle

* Comment out system tests
AddCommandSystemTest
ClearCommandSystemTest
DeleteCommandSystemTest
EditCommandSysetmTest
ErrorDialogGuiTest
FindNameCommandSystemTest

* Revert "Comment out system tests"

This reverts commit ff7e67975b6a73072eba1a4bdd30a068a9dc7eca.

* Comment out tests
AddCommandSystemTest
ClearCommandSystemTest
DeleteCommandSystemTest
EditCommandSystemTest
ErrorDialogGuiTest
FindNameCommandSystemTest

* Fix checkstyle

* Revert "Fix checkstyle"

This reverts commit 63f8633d9f10e90d8d6a04e4a9ba28c3ad4e7d60.

* Revert "Comment out tests"

This reverts commit 7f30caf0534033c830136577fb956bf99e22f9c1.

* Comment out system tests.
AddCommandSystemTest
ClearCommandSystemTest
DeleteCommandSystemTest
ErrorDialogGuiTest
FindNameCommandSystemTest
SelectCommandSystemTest

* Comment out system tests #2

* Revert "Comment out system tests #2"

This reverts commit c38348072486f7281617d84c1fb787fb709f8855.

* Comment out system tests #2

guekling committed:
[#171] Fixes App Loading Too Slow (#176)

* Solve the app loading too slow problem

* Fix ClearCommandSystemTest

* Fix EOF issue

* Delete blank line

guekling committed:
[#179] Fixes Calendar Tests Fails (#180)

* Fix JavaFX thread issues

* Fix MonthViewTest issues

* Uncomment PreviousMonthCommandTest

* Fix CalendarPanelTest issues

guekling committed:
[#173] Modify DeveloperGuide - Introduction, Getting Started & Design (#172)

* Add Introduction

* Modify 'Setting up the project'

* Modify 'Configuring the project'

* Delete 'Updating documentation'

* Add 'Getting started with developing'

* Modify asciidoc stylesheets

* Add linkcss & font @ header for HTML

* Modify 'Architecture'

* Modify 'Design'

* Edit padding-bottom for headings

* Empty commit to netlify

* Add CSS stylesheets

* Attempt to solve MonthViewTest.equals() random fails

guekling committed:
[#181] Add NextMonthCommand (#183)

* Add NextMonthCommand

* Add NextMonthCommandTest

* Add more test to PreviousMonthCommandTest

* Add tests in MonthViewTest

* Update UserGuide

* Update OrganizerParserTest

* Increase pause time

dominickenn committed:
[#143][#173] Developer Guide edit (#184)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Edit UML diagrams

* Edited implementation

* Edit logging

* Edit configuration

* Edit documentation

* Edit testing

* Edit dev ops

* Edit appendix A

* Edit appendix B

* Edit appendix C

* Minor edit appendix F

* Fix checkstyle

guekling committed:
[#174] Rework User Guide: Layout Structure, Introduction, Quick Start & Using PrioriTask & FAQ (#190)

* Modify Introduction

* Minor bullet point modification to DeveloperGuide

* Remove DummySearchPage

* Update QuickStart, Reworking layout of UserGuide (in progress)

* Rework entire UserGuide

* Add section Using PrioriTask

* Cleanup of layout, Adding of placements

* Modify FAQ

* Add HTML file

* Re-rendered HTML file

* Edit typo

* Add EOF

* Re-rendered HTML

* Delete UserGuide.html due to EOF issues

guekling committed:
[#182] Add CurrentMonthCommand (#191)

* Add CurrentMonthCommand

* Fix typo error

* Modify getExpectedDateRow

* Add tests

* Update UserGuide

* Update version number

* Change AddressBook -> PrioriTask in Congif.java

* Edit credits for collate

* Is Travis failing randomly?

* Fix ConfigTest

aguss787 committed:
Fix retrieving subtask from wrong source (#175)

* Fix retrieving subtask from wrong source

* Fix testcase

dominickenn committed:
[#186][#187][#188] Add ForgotPasswordCommand and AnswerCommand (#193)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Uncomment tests

* Fix user typo

* Add UserWithQuestionAnswer class

* Add AddQuestionAnswerCommand and edit user structure

* Add tests

* Fix checkstyle

* Add getUserByUsername method

* Add forgotPasswordCommand

* Add tests

* Add answerCommand

* Add tests

* Fix checkstyle and add documentation

* Comment out system tests

* Fix checkstyle

guekling committed:
Collate code, Fix collate code issues (#196)

dominickenn committed:
[#189][#194][#151][#128] Add tests and update UML diagram (#197)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Add tests

* Update UML diagram

* Fix checkstyle

guekling committed:
[#173] Rework Developer Guide: Undo/Redo Implementation (#198)

* Add syntax highlighting

* Remove undo/redo tag

* Fix Undo/Redo section issues

* Add fullstops & TODO

* Add more TODOs

* Add TODO

aguss787 committed:
Add collated code (#199)

dominickenn committed:
[#174] Update TODO and language check. (#200)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Update UserGuide

* Fix stylecheck

guekling committed:
[#174] User Guide: Update Calendar Features (#202)

* Edit preface for Calendar Features

* Edit README

* Fix checkstyle issue

guekling committed:
[#132] DeveloperGuide: Add Calendar Implementation (#201)

* Added comment

* Update calendar implementation

* Edit HTML link

dominickenn committed:
[#203] Fix logout bug (#204)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Fix bug and add tests

guekling committed:
UGDG: Resolve Minor Issues (#207)

* Fix minor issues

* Fix minor issues in DevGuide

dominickenn committed:
[#47][#49][#205][#206] Add task filtering commands (#208)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Uncomment system tests

* Add TaskByStatusPredicate

* Add test for new predicate

* Add ListUncompletedTasksCommand

* Add test for new command

* Refactor names

* Add command to UG

* Add ListCompletedTasksCommand

* Add tests

* Add documentation in UG

* Add uncommented system tests folder

* Remove folder

* Fix stylecheck

* Comment out system tests

* Add folder of system tests (uncommented version)

brandonyeoxg commented on src/main/java/seedu/organizer/logic/commands/AddQuestionAnswerCommand.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    private UserWithQuestionAnswer createEditedUser() {
+        requireNonNull(userToEdit);
+        UserWithQuestionAnswer user = new UserWithQuestionAnswer(
```
We could shorten this code by a little by doing:
``` java
    return new UserWithQuestion(...)
```

brandonyeoxg commented on src/test/java/seedu/organizer/logic/commands/AddQuestionAnswerCommandTest.java:
```diff
@@ -0,0 +1,93 @@
+    }
+
+    @Test
+    public void equals() {
```
Recall from our tutorials a few weeks back on how we can approach testing our method, we could split it into several smaller tests so that it is easier for us to pin point which combination of inputs or equivalence partitions are causing failed cases.

brandonyeoxg commented on src/test/java/seedu/organizer/logic/commands/AnswerCommandTest.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    @Test
+    public void execute_existingUser_answer() throws Exception {
```
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. name doesn't follow naming convention

Recall that the last world of the method name should be the <expectedBehaviour>

brandonyeoxg reviewed (commented):
@dominickenn @guekling @aguss787 @natania-d hey guys just some quick unsolicited comments. 
Overall the team is on track with v1.4 and the coding standards are relatively good. 
The comments that I've made are just there to improve it a little bit more =).
Keep it up for v1.5!

dominickenn committed:
[#210] Fix App does not appear bug (#211)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Fix window coordinate bug

* Fix window size bug

* Fix stylecheck

dominickenn committed:
[#213] Fix login bug (#215)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Fix login bug

* Add tests

* Restore HelpCommandSystemTest

dominickenn committed:
[#233][#216] Delete select command (#235)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Delete SelectCommand

* Comment out system tests

* Remove select from UG

guekling committed:
Fix trailing & leading spaces bug (#237)

dominickenn committed:
[#217][#219][#221][#222][#231][#236] Fix login feature related bugs. (#239)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Fix username and password length bug

* Added documentation for username/password requirements

* Fix history command leak information bug

* Fix undoable signup command

* Add documentation for forgetpassword and ans bug.

* Comment out system tests

guekling committed:
[#234] Add isValidDate to fix deadline bugs (#240)

* Fix copy & paste errors

* Warning issued when date is invalid

* Remove all occurences of Select

* Fix copy+paste issues

* Update UserGuide

* Typo fix

guekling committed:
[#232][#218] Update UserGuide to Reflect Deadline Parameter Constraints (#244)

* Update UserGuide

* Attempt at fixing the screen size issue

dominickenn committed:
[#242] Rework priority autoupdating formula. (#243)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Update formula

* Add documentation

* Fix stylecheck

* Update formula

* Add documentation

* Fix stylecheck

dominickenn committed:
[#226][#246] Add UI to replace total tasks with currently logged in user. (#247)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Replace total tasks status with current user status

* Fix tests

* Replace uncommentedSystemTest

* Comment out system tests

* Uncomment organizer system test

dominickenn committed:
[#250][#251] Fix typo in XmlAdaptedTask and DateCompleted (#252)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

dominickenn committed:
[#224][#228] Edit MESSAGE_USAGE (#253)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Change "Done" to "Completed"

* Edit message usage

dominickenn committed:
[#248] Edit DG (everything except appendices) (#254)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Section 1 and 2.

* Section 3.1

* Section 3.2

* Section 3.3

* Section 3.4

* Section 3.5

* Section 3.5

* Section 4

* Section 5

* Section 6

* Section 7

dominickenn committed:
Code style check and Project Portfolio for Dominic (V1.0) (#256)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Collate code

* Functional code codestyle rework

* Test code codestyle rework

* PP edit #1

* Comment out tests

* Fix stylecheck

guekling committed:
Update Project Portfolio Page (#258)

* Delete example PPP

* Updated my PPP

* Include NOF newline

natania-d committed:
Create RecurWeeklyCommand without logic yet (#192)

* Create Recurrence.java

* Create XmlAdaptedRecurrence.java

* Refactor to include Recurrence in Task constructors in other classes

* Create RecurWeekly Command

* Fixed checkstyle errors.

* Fixed checkstyle errors

* Fixed checkstyle error

* Fixed compileTest error

* Implement Recur Weekly Command (incomplete)

* Implement Recur Weekly command with no tests yet

* Fix bugs in tests

* Fix checkstyle error

* Update User Guide.

* Fix compileTestJava error

* Fix checkstyleTest error

natania-d committed:
Project Portfolio for natania (#259)

* Start on Project Portfolio for natania

* Fix error

* Fix error

* Update tag for User Guide in natania Project Portfolio

aguss787 committed:
Add portfolio (#257)

* Add portfolio

* Added newline

guekling committed:
Fix calendar size issue (#260)

dominickenn committed:
Add command parser test bug fix (#261)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Bug fix

* Fix checkstyle

guekling committed:
[#223] Fix Multiple Prefixes for Task Commands (#249)

* Another attempt at fixing the resizing

* Fix multiple same prefixes issue

* Add comments

* Fix copy + paste issues

* Fix EditCommand add multiple same prefixes

* Remove useless tests in EditCommandParserTest

* Update tests for AddCommandParserTest

* Fix Travis failures

* Add tests for EditCommandParserTest

* Update UserGuide

* Change message result

* Fix multiple same prefixes, and update tests

* Fix trailing whitespace?

* Empty commit

dominickenn committed:
Update Dominic's Project Portfolio (#262)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Update Dominic's Project Portfolio

aguss787 committed:
Add edit subtask (#255)

* Add edit subtask

* Add test

* Fix formatting

* Add missing author tag

* Hot fix

* Add basePriority and recurrence

guekling committed:
Modify PrioriTask's UI (#263)

* Edit theme style

* Change the layout of MainWindow.fxml

* Fix build error

* Add new logo

* Update Ui.png and README

* Fix test fail

aguss787 committed:
Hotfix for UG (#264)

* Fix tag

* Formatting

* Edit subtasks commands

brandonyeoxg commented on src/main/java/seedu/organizer/model/subtask/UniqueSubtaskList.java:
```diff
@@ -129,4 +168,14 @@ protected DuplicateSubtaskException() {
+     */
+    public void makeAllSubtasksUndone(List<Subtask> subtasks) {
+        for (Subtask subtask : subtasks) {
+            ;
```
Hey guys if the feature cannot be implemented in time, then perhaps it would be wise to remove this piece of code and update the user guide on the limitations.

brandonyeoxg commented on src/main/java/seedu/organizer/model/task/Deadline.java:
```diff
@@ -38,7 +46,32 @@ public Deadline(String deadline) {
+            sdf.setLenient(false);
+            Date date = sdf.parse(test);
+
+            if (date.compareTo(new Date()) < 0) {
```
Is this line really required? 
It seems that the method will return true even if `data.compareTo(new Date()) >= 0` as per line 65.

brandonyeoxg reviewed (commented):
@dominickenn @guekling @aguss787 @natania-d some quick comments.
Overall from my random sampling, the code quality is pretty good.

I'd advise against implementing any new features, instead focus on pure bug fixing and writing test cases. (Pretty sure you guys are already doing it =P).

Remember from tutorial on bringing along your adapters to connect to the monitors so that we have redundancy of laptops incase one fails.

DG and UG content wise should not be a problem, if you want to improve on them then perhaps adding screenshots would help. Also do remember to **update** all your existing screenshots,

Do remember to collate your codes.

Finally all the best for your upcoming presentation!

dominickenn committed:
[#248] Add to Appendix F of DG (#265)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Edit glossary

* Add user management manual testing

* Fix code bugs

* Add note for window size resizing

* Add markings for group mates to work on

* Fix stylecheck

guekling committed:
Fix UGDG Issues (#267)

* Update Introduction

* Updating Round 1

* Update Round 2

natania-d committed:
Implement DeleteRecurredTasksCommand (#266)

* Start on DeleteRecurredTasksCommand

* Complete implementation of DeleteRecurredTasksCommand

* Add tests for DeleteRecurredTasksCommand

* Fix checkstyle errors

dominickenn committed:
Edit UG inconsistency. (#270)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Edit UG inconsistencies

natania-d committed:
Implement RemoveTagsCommand (#269)

* Implement RemoveTagsCommand

* Add tests for RemoveTagsMethod

* Fix checkstyle errors

* Fix checkstyle errors

* Start adding UG documentation for RemoveTagsCommand

* Finish documentation in UG of RemoveTagsCommand

* Fix error for tests

guekling committed:
Remove Unnecessary Code, Update Docs (#271)

* Delete unnecessary code in Deadline

* Update PPP

* Update Appendix F in DevGuide

* Made minor changes to DevGuide

* Fix checkstyle error

* Minor changes to DevGuide

* CAPS

aguss787 committed:
Add some manual tc (#272)

guekling committed:
Collate Code, Update Docs (#273)

* Edit PPP

* Collate

* Edit UG Summary

natania-d committed:
DG and UG improvements (#274)

* Update developer guide and user guide

* Add to appendix F of DG

* Update documentation

* Fix checkstyle error

* Revert "Fix checkstyle error"

This reverts commit bb31698cd7b5e1cbd190ea98aa6e059aa97c71d1.

* Fix checkstyle error

dominickenn committed:
Update Dominic's PPP (#275)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Update PPP

* Update number of PRs reviewed

dominickenn committed:
Update version number to v1.5 (#276)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Edit version number

dominickenn committed:
Update Dominic's PPP. (#278)

* Add major/minor feature documentation for everyone

* Userguide typo fixes

* Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main

# Conflicts:
#	src/main/java/seedu/organizer/logic/parser/OrganizerParser.java
#	src/test/java/systemtests/AddCommandSystemTest.java
#	src/test/java/systemtests/EditCommandSystemTest.java
#	src/test/java/systemtests/FindNameCommandSystemTest.java

* Revert "Merge branch 'LoginTestDevelopment' of https://github.com/dominickenn/main"

This reverts commit a55ef3e16d15e877b55d8374e5d951c4dff07c01.

* Fix merge conflict

* Edit message

* Typo error fix

* Update TestApp.java

* Update AddCommandParserTest.java

* Remove documentation.

natania-d committed:
Update documentation (#277)

* Update documentation

* Work on UG

* Update documention. Add table for UG for Command Summary.

natania-d committed:
Change to correct Github username to collate code for natania-d and aguss787 (#279)

natania-d committed:
Change link from natania to natania-d
